### DISCUSSION TITLE
Issues with Convergence when Switching from PH_case to naca0012 in Field Inversion

### DISCUSSION THREAD

**Discussion by lnhang0618** [mdolab/dafoam#448] [General]
Dear Prof. He,

I am currently working on a field inversion example, transitioning from the PH_case to naca0012. I used [runTests](https://github.com/mdolab/dafoam/blob/main/tests/runTests_DASimpleFoamFieldInversionObjectiveFunctions.py) and the [naca0012 tutorial](https://github.com/DAFoam/tutorials/tree/main/NACA0012_Airfoil/incompressible) as references for this process.

I replaced the FFD/wingFFD，0.orig and constant/polyMesh files with those specific to naca0012. Additionally, I adapted the UData file in accordance with the test settings, utilizing varRefFieldInversion as shown in lines 24 and 30 of the [runTests](https://github.com/mdolab/dafoam/blob/main/tests/runTests_DASimpleFoamFieldInversionObjectiveFunctions.py) script.

As for the 'runScripts', I've retained the vast majority of the settings, with the exception of removing the 'fvsource' item and modifying the global parameters. This means that the 'objfunc' part is identical to the one in the PH_case, with the inclusion of UData in the loss function.

Despite my efforts, the velocity field 'U', has not converged towards UData as anticipated.I'm hoping to gain insight into potential issues or mistakes I might be making in this process.

**Comment by lnhang0618** [comment]
Here is the zip file.[testfile.zip](https://github.com/lnhang0618/testfile/archive/refs/heads/main.zip)

**Reply by uttamcadambi07** [reply]
Have you managed to find a solution to this?

**Reply by lnhang0618** [reply]
Hi uttam，
Sorry，I haven‘t fix it yet since I have to pay most of my time on another work.But omid have given me a reference as well as some advisable suggestions.I have forwarded the email to your inbox.

**Reply by uttamcadambi07** [reply]
Thanks a lot I have received the email.

**Comment by friedenhe** [comment]
Did the flow and adjoint converge well?
